src/storage-controller/src/collection_mgmt.rs

```diff
@@ -232,7 +255,20 @@ where
                         if let Some(batch) = cmd {
                             // To rate limit appends to persist we add artifical latency, and will
                             // finish no sooner than this instant.
-                            let min_time_to_complete = Instant::now() + DEFAULT_TICK;
+                            let batch_duration_ms = match id {
+                                GlobalId::User(_) => Duration::from_millis(user_batch_duration_ms.load(Ordering::Relaxed)),
+                                // For non-user collections, always just use the default.
+                                _ => STORAGE_MANAGED_COLLECTIONS_BATCH_DURATION_DEFAULT,
+                            };
+                            let use_batch_now = Instant::now();
+                            let min_time_to_complete = use_batch_now + batch_duration_ms;
```

Comment:
In `mz_storage_controller::statistics` I used a `watch` to communicate the new interval and just overrode the `interval` variable, but for this I think this is better!

You do need to override the `inteval` with a new one with the new duration, if its changed though, right?